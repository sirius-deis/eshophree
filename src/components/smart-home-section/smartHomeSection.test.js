import { render, screen } from "@testing-library/react";
import SmartHomeSection from "./smartHomeSection";
import { __esModule } from "@testing-library/jest-dom/dist/matchers";

jest.mock('../card/card.jsx', () => ({
  __esModule: true,
  default: ({name, price}) => (
    <div>
      <h2>{name}</h2>
      <p>{price}</p>
    </div>
  )
}))

jest.mock('../../hooks/useFetch.js', () => ({
  __esModule: true,
  default: jest.fn()
}))

describe("SmartHomeSection component", () => {
  const mockProducts = {
    products: [
      { _id: 1, name: "Product 1", price: 100 },
      { _id: 2, name: "Product 2", price: 200 },
    ]
  };
  beforeEach(() => {
    jest.clearAllMocks();
  })
  it("should match snapshot", () => {
    const useFetch = require('../../hooks/useFetch').default;
    useFetch.mockReturnValue([mockProducts, false]);
    const {container} = render(<SmartHomeSection />);
    expect(container).toMatchSnapshot();
  });
  it("should render loading state initially", () => {
    const useFetch = require('../../hooks/useFetch').default;
    useFetch.mockReturnValue([null, true]);

    render(<SmartHomeSection />);
    
    expect(screen.queryByText("Smart Home Appliances")).not.toBeInTheDocument();
  });
  it("should render products", () => {
    const useFetch = require('../../hooks/useFetch').default;
    useFetch.mockReturnValue([mockProducts, false]);

    render(<SmartHomeSection />);

    mockProducts.products.forEach(product => {
      expect(screen.getByText(product.name)).toBeInTheDocument();
      expect(screen.getByText(product.price)).toBeInTheDocument();
    })
  });
});