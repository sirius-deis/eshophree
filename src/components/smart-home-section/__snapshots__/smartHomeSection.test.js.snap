// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SmartHomeSection component should match snapshot 1`] = `
<div>
  <section>
    <div
      class="sc-fFGjHI dwKuma"
    >
      <div
        class="sc-ewnqHT dLOUny"
      >
        <div
          style="display: flex; align-items: center; gap: 0.7rem;"
        >
          <div
            class="sc-gTRrQi eoyTqP"
          >
            Smart Home Appliances
          </div>
        </div>
        <div>
          <button
            class="sc-gxYJeL sc-kPTPQs kCKvOC"
            disabled=""
          >
            ‹
          </button>
          <button
            class="sc-gxYJeL sc-ispOId kCKvOC"
            disabled=""
          >
            ›
          </button>
        </div>
      </div>
      <div
        class="sc-bgqQcB iPXChy"
        width="100"
      />
      <div
        style="display: flex; gap: 1rem;"
      >
        <div
          class="sc-dcrmVg bOxfiE"
        />
        <div
          style="overflow: hidden;"
        >
          <div
            class="sc-dicizt cTmuUj"
            style="transform: translateX(100%);"
          />
        </div>
      </div>
    </div>
  </section>
</div>
`;
